File: Java/Góry i doliny.java
import java.util.Scanner;

public class Main
{
    public static void main(String[] args) 
    {
        int increase = 0;
        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        int[] heights = new int[n];
        for (int i = 0; i < n; i++)
            heights[i] = input.nextInt();
        for (int i = 1; i < n - 1; i++)
            if (heights[i - 1] < heights[i] && heights[i] > heights[i + 1])
                increase++;
        System.out.print(increase);
        input.close();
    }
}

File: C++/Góry i doliny.cpp
#include <vector>
#include <iostream>

int main() 
{
    int n, increase = 0;
    std::cin >> n;
    std::vector<int> heights(n);
    for (int i = 0; i < n; i++)
        std::cin >> heights[i];
    for (int i = 1; i < n - 1; i++)
        if (heights[i - 1] < heights[i] && heights[i] > heights[i + 1])
            ++increase;
    std::cout << increase;
    return 0;
}